tests/js/common/shell/test-kill-queries.js

```diff
@@ -0,0 +1,328 @@
+  }
+  testCases.push(createTestCaseEntry("Query::directKillBeforeQueryWillBeFinalized", false, 'both', "both"));
+  testCases.push(createTestCaseEntry("Query::directKillAfterQueryWillBeFinalized", false, 'both', "both"));
+  testCases.push(createTestCaseEntry("Query::directKillAfterDBServerFinishRequests", true, 'both', "both"));
+
+  const testSuite = {
+    setUp: function () {
+      db._useDatabase("_system");
+      db._createDatabase(databaseName);
+      db._useDatabase(databaseName);
+
+      db._create(collectionName, {numberOfShards: 4});
+    },
+
+    tearDown: function () {
+      console.warn("CALLING TEAR DOWN, DROPPING DB");
+      db._useDatabase("_system");
+      db._dropDatabase(databaseName);
+    },
+  };
+
+  const createTestName = (failurePoint, stream) => {
+    let typedef = '_';
+    if (stream === 'on') {
+      typedef = '_stream_';
+    } else if (stream === 'off') {
+      typedef = '_nonstream_';
+    }
+    return `test${typedef}${failurePoint.split("::")[1]}`;
+  };
+
+  const addTestCase = (suite, testCase) => {
+    if (!internal.isCluster() && testCase.onlyInCluster) {
+      return;
+    }
+
+    suite[createTestName(testCase.failurePointName, testCase.stream)] = function (failurePointName, stream, reportKilled) {
+      console.warn("Failure Point: " + failurePointName);
+      console.warn("Stream type: " + stream);
+
+      try {
+        internal.debugSetFailAt(failurePointName);
+      } catch (e) {
+        // Let the Test fail
+        throw `Failed to initialize failurepoint ${failurePointName}`;
+      }
+
+      try {
+        if (stream === 'on') {
+          executeStreamCursorQuery(reportKilled);
+        } else if (stream === 'off') {
+          executeDefaultCursorQuery(reportKilled);
+        } else if (stream === 'both') {
+          executeStreamCursorQuery(reportKilled);
+          executeDefaultCursorQuery(reportKilled);
+        }
+      } finally {
+        try {
+          internal.debugClearFailAt(failurePointName);
+        } catch (e) {
+          // We cannot throw in finally.
+          console.error(`Failed to erase debug point ${failurePointName}. Test may be unreliable`);
+        }
+      }
+
+    }.bind(this, testCase.failurePointName, testCase.stream, testCase.reportKilled, testCase.onlyInCluster);
+  };
+
+  for (const testCase of testCases) {
+    addTestCase(testSuite, testCase);
+  }
+
+  // add two regular tests (one default, one stream) without breakpoint activation
+  testSuite["test_positiveStreamQueryExecution"] = function () {
+    let localQuery = db._query(exlusiveWriteQueryString, null, null, {stream: true});
+    assertTrue(localQuery);
+    localQuery.dispose();
+  };
+
+  testSuite["test_positiveDefaultQueryExecution"] = function () {
+    let localQuery = db._query(exlusiveWriteQueryString);
+    assertTrue(localQuery);
+  };
+
+  if (!isServer && !internal.isCluster()) {
+    testSuite["test_export_api"] = function () {
```

Comment:
This will very likely fail once PR https://github.com/arangodb/arangodb/pull/13929 is merged.